Packages/Signer/Sources/Chains/HyperCoreSigner.swift

```diff
@@ -5,40 +5,30 @@ import Blockchain
 import Foundation
 import Gemstone
 import GemstonePrimitives
-import Keychain
-import Keystore
 import Primitives
 import WalletCore
 import WalletCorePrimitives
 
 public class HyperCoreSigner: Signable {
-    let keychain: Keychain
-    let hyperCore = HyperCore()
-    let factory = HyperCoreModelFactory()
+    private let preferences: HyperliquidSecurePreferences
+    private let hyperCore = HyperCore()
+    private let factory = HyperCoreModelFactory()
     private let agentNamePrefix = "gemwallet_"
     
-    public init(keychain: Keychain = KeychainDefault()) {
-        self.keychain = keychain
+    public init(preferences: HyperliquidSecurePreferences) {
+        self.preferences = preferences
     }
 
     public func signTransfer(input: SignerInput, privateKey: Data) throws -> String {
         throw AnyError.notImplemented
     }
 
-    func getAgentKey(for walletAddress: String) throws -> (address: String, key: Data) {
-        let agentPrivateKeyName = "\(HyperCoreService.agentPrivateKey)_\(walletAddress)"
-        let agentAddressKeyName = "\(HyperCoreService.agentAddressKey)_\(walletAddress)"
-        
-        if let key = try keychain.get(agentPrivateKeyName), let address = try keychain.get(agentAddressKeyName) {
-            return try (address: address, Data.from(hex: key))
+    func getHyperliquidKey(for walletAddress: String) throws -> HyperliquidKey {
```

Comment:
```suggestion
    func getAgentKey(for walletAddress: String) throws -> HyperliquidKey {
```

I would use as agent